app/models/audio_version.rb

```diff
@@ -20,4 +25,33 @@ def length(reload=false)
   def self.policy_class
     StoryAttributePolicy
   end
+
+  def compliant_with_template?
+    file_errors.nil?
+  end
+
+  private
+
+  def validate_on_template
+    return unless audio_version_template
+
+    noncompliant_file = audio_files.find do |af|
+      !af.compliant_with_template? || af.status == INVALID
```

Comment:
`compliant_with_template` currently checks only the `audio_errors`. So for now, this line is looking for either a file that's uploaded successfully but doesn't comply with the template, or a file has uploaded unsuccessfully & been deemed invalid by the lambda callback. 

as for whether it should account for notfound/ validating/ failed -- could make sense to do so! what do you think?